core/config/project_settings.cpp

```diff
@@ -609,6 +609,24 @@ Error ProjectSettings::_setup(const String &p_path, const String &p_main_pack, b
 			// Attempt to load PCK from macOS .app bundle resources.
 			found = _load_resource_pack(OS::get_singleton()->get_bundle_resource_dir().path_join(exec_basename + ".pck")) || _load_resource_pack(OS::get_singleton()->get_bundle_resource_dir().path_join(exec_filename + ".pck"));
 		}
+
+		if (!found) {
+			// Attempt to load project file from macOS .app bundle resources.
+			String bundle_resource_dir = OS::get_singleton()->get_bundle_resource_dir();
```

Comment:
`get_bundle_resource_dir` can return empty string is app is not bundled, so it should check if it's not empty. For consistency, I would also move it down to the part after `// Try to use the filesystem for files, according to OS.`.